test/compile/scale/init.test.ts

```diff
@@ -35,5 +35,13 @@ describe('compile/scale', () => {
       assert.equal(scale.paddingOuter, 0.15);
       assert.isUndefined(scale.padding);
     });
+
+    it('should have scale type as utc if timeunit is utc', () => {
+      const scale = initScale(
+        'x', {field: 'a', type: 'temporal', scale: {type: 'band'}, timeUnit: 'utcyear'}, {scale: {bandPaddingInner: 0.3}},
```

Comment:
This seems wrong.   If users specify band, then it should use band?

We should use utc if  (1) no scale type is specified or (2) specified scale type is not supported.  

(Maybe `"band"` is not supported -- but given the test as is, this is slightly confusing.)